Add GDAL, version 1.11.1_3

gmisra opened the pull request:
GDAL (http://www.gdal.org/) is a library package, and some of the projects that depend on it are version-sensitive, so being able to explicitly install a specific version should be helpful to others.

gmisra commented:
Follow-up from PR #588

DomT4 commented on gdal110.rb:
```diff
@@ -0,0 +1,425 @@
+require 'formula'
```
:scissors: this

DomT4 commented on gdal110.rb:
```diff
@@ -0,0 +1,425 @@
+require 'formula'
+
+class Gdal110 < Formula
+  homepage 'http://www.gdal.org/'
```
Double quotes around everything.

DomT4 commented on gdal110.rb:
```diff
@@ -0,0 +1,425 @@
+  sha1 'b4df76e2c0854625d2bedce70cc1eaf4205594ae'
+
+  option 'complete', 'Use additional Homebrew libraries to provide more drivers.'
+  option 'enable-opencl', 'Build with OpenCL acceleration.'
```
All of these options should be ` with-z ` or ` without-z `.

DomT4 commented on gdal110.rb:
```diff
@@ -0,0 +1,425 @@
+  depends_on "postgresql" => :optional
+  depends_on "mysql" => :optional
+
+  depends_on 'homebrew/science/armadillo' if build.include? 'enable-armadillo'
```
Just make this ` => :optional `.

DomT4 commented on gdal110.rb:
```diff
@@ -0,0 +1,425 @@
+  end
+
+  resource 'numpy' do
+    url 'http://downloads.sourceforge.net/project/numpy/NumPy/1.8.1/numpy-1.8.1.tar.gz'
```
` https:// `

DomT4 commented on gdal110.rb:
```diff
@@ -0,0 +1,425 @@
++    #if defined(__APPLE__) && (!defined(TARGET_OS_IPHONE) || TARGET_OS_IPHONE==0)
+         #include <crt_externs.h>
+         #define environ (*_NSGetEnviron())
+     #else
```
File must end with a new blank line.

DomT4 commented on gdal111.rb:
```diff
@@ -0,0 +1,310 @@
+require 'formula'
```
Same comments apply to this formula as the one above.

DomT4 commented:
Gonna ping @mikemcquaid here for general thoughts/feelings, and @tdsmith for any thoughts on the Python elements. Thanks in advance Mike/Tim! :heart:

DomT4 commented on gdal110.rb:
```diff
@@ -0,0 +1,425 @@
+    end
+
+    cd 'swig/python' do
+      system "python", "setup.py", "install", "--prefix=#{prefix}", "--record=installed.txt", "--single-version-externally-managed"
```
This needs to use Homebrew's new Python arguments. See [jrnl.rb](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/jrnl.rb) upstream for an example.

tdsmith commented:
I need to fix this in Homebrew/homebrew as well but it would be polite to add a helper like

```ruby
  def package_installed? python, module_name
    quiet_system python, "-c", "import #{module_name}"
  end
```

and only install numpy into `prefix` if `!package_installed?("python", "numpy")`

mikemcquaid commented:
Seems fine to me :+1:

DomT4 commented:
Cool. @gmisra If you address my comments, anything ` brew audit --strict ` has to say beyond that, and work with Tim on the Numpy issue I'm happy to merge this.

gmisra commented:
I will address these comments within the next few days, thanks again for all your help.

gmisra commented:
@DomT4 changes made.

DomT4 commented:
Thanks! Part of the patch is failing now - It might be worth sticking it in a gist rather than inline. The inline patches can be a little temperamental.

```
==> Verifying gdal110-1.10.1.tar.gz checksum
tar xf /Library/Caches/Homebrew/gdal110-1.10.1.tar.gz
==> Patching
patching file GDALmake.opt.in
Hunk #1 FAILED at 123.
1 out of 1 hunk FAILED -- saving rejects to file GDALmake.opt.in.rej
patching file configure
Hunk #2 FAILED at 23046.
1 out of 3 hunks FAILED -- saving rejects to file configure.rej
patching file configure.in
Hunk #1 FAILED at 2294.
1 out of 2 hunks FAILED -- saving rejects to file configure.in.rej
patching file ogr/ogrsf_frmts/mysql/GNUmakefile
Hunk #1 FAILED at 7.
1 out of 1 hunk FAILED -- saving rejects to file ogr/ogrsf_frmts/mysql/GNUmakefile.rej
patching file ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp
patching file port/cpl_spawn.cpp
```

Couple of other audit things:

```
==> brew style gdal110
== /usr/local/Library/Taps/homebrew/homebrew-versions/gdal110.rb ==
C:200: 25: Use def with parentheses when there are parameters.
C:205: 29: Use && instead of and.
```

And some for the other formula as well:

```
==> Patching
/usr/bin/patch -g 0 -f -p2 -i /private/tmp/gdal111--patch-Jyyubx/12d3b984a052c59ee336f952902b82ace01ba31c.diff
patching file ogr/ogrsf_frmts/sqlite/ogrsqlitevirtualogr.cpp
Error: inreplace failed
configure:
  expected replacement of /^mandir="\$\{prefix\}\/man"$/ with ""
==> brew audit Homebrew/versions/gdal111 --strict			 FAILED
==> brew style gdal111
== /usr/local/Library/Taps/homebrew/homebrew-versions/gdal111.rb ==
C:224: 25: Use def with parentheses when there are parameters.
C:229: 29: Use && instead of and.

1 file inspected, 2 offenses detected
```

The inreplace and patch failures are new. There may have been some spacing/quotation marks errors created on those. Might be worth another look over.

Appreciate the small mountain of failures doesn't look like progress, but nearly there on this one.

@tdsmith Python changes look okay to you?

gmisra commented:
I fixed the style issues from brew audit, and fixed the failing inreplace (the upstream source changed).

I don't really understand how the patch section works, can you point to an example of how the patch information can be stored in a gist?

DomT4 commented:
Sure: [Gist patch example](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/glog.rb#L20-L23).

DomT4 commented:
@BrewTestBot test this please

DomT4 commented:
Ping?

gmisra commented:
Have not forgotten, have been busy, but it's on the todo list.


On Wed, Apr 1, 2015 at 9:43 PM, Dominyk Tiller <notifications@github.com>
wrote:

> Ping?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew-versions/pull/675#issuecomment-88732758>
> .
>

DomT4 commented:
Thanks. Just don't like to leave stuff sitting too long without getting an update on it.

DomT4 commented:
@BrewTestBot test this please

gmisra commented:
@DomT4 I just spent some quality time learning how homebrew handles patching, and I am stumped by these patch failures. I tried recreating each patch from scratch, following the directions in the cookbook (in interactive mode), and I am still seeing failures, even for a single patch that is just inserting one line. Any tips on how to investigate further?

Example:
- For this formula https://github.com/gmisra/homebrew-versions/blob/gdal_versions/gdal110.rb, the first patch just inserts one line (L280).
- I deleted all of the patch data from the formula, installed the formula interactively, edited that file to add one line, and used `git diff | pbcopy` to copy the patch data back into the formula.
- When I rerun the formula, it still fails on the same patch, at the same point.

DomT4 commented:
Yeah, the patch isn't getting applied cleanly, to say the least. Where are you getting the patch from? It looks the the original patch itself may either be outdated for the codebase, or has copied well across formats. If you wrote the patch yourself, I'm not sure why it wouldn't apply.

gmisra commented:
I did not write the patch, I grabbed it from the homebrew repository's commit history (https://github.com/Homebrew/homebrew/blob/4c3c8defc77bc379df684fe73e1c909975c15581/Library/Formula/gdal.rb). The checksum of the source file hadn't changed, so I would assume the file being patched hasn't changed either. But I'm pretty stumped.

DomT4 commented:
Apparently, there are some rogue spacing fixes in your ` __END__ ` block that don't exist in the original. Red is you, green the original.

```diff
diff --git a/gdal110.rb b/gdal110.rb
index 0b14df8..684c054 100644
--- a/gdal110.rb
+++ b/gdal110.rb
@@ -274,11 +274,11 @@ index d7273aa..2fcbd53 100644
 --- a/GDALmake.opt.in
 +++ b/GDALmake.opt.in
 @@ -123,6 +123,7 @@ INGRES_INC = @INGRES_INC@
- HAVE_MYSQL = @HAVE_MYSQL@
- MYSQL_LIB  = @MYSQL_LIB@
- MYSQL_INC  = @MYSQL_INC@
+ HAVE_MYSQL =	@HAVE_MYSQL@
+ MYSQL_LIB  =	@MYSQL_LIB@
+ MYSQL_INC  =	@MYSQL_INC@
 +MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION  =    @MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION@
- LIBS    += $(MYSQL_LIB)
+ LIBS	   +=	$(MYSQL_LIB)
  
  #
 diff --git a/configure b/configure
@@ -325,7 +325,7 @@ index 1c4f8fb..120b17f 100755
 +      rm -f testmysql.*
 +      rm -f testmysql
 +
-  ;;
+ 	;;
    esac
  fi
 @@ -23055,6 +23084,8 @@ MYSQL_INC=$MYSQL_INC
@@ -370,7 +370,7 @@ index 481e8ea..d83797f 100644
 +      rm -f testmysql.*
 +      rm -f testmysql
 +
-  ;;
+ 	;;
    esac
  fi
 @@ -2301,6 +2326,7 @@ fi
@@ -385,16 +385,16 @@ diff --git a/ogr/ogrsf_frmts/mysql/GNUmakefile b/ogr/ogrsf_frmts/mysql/GNUmakefi
 index 292ae45..e78398d 100644
 --- a/ogr/ogrsf_frmts/mysql/GNUmakefile
 +++ b/ogr/ogrsf_frmts/mysql/GNUmakefile
-@@ -7,6 +7,11 @@ OBJ  = ogrmysqldriver.o ogrmysqldatasource.o \
+@@ -7,6 +7,11 @@ OBJ	=	ogrmysqldriver.o ogrmysqldatasource.o \
  
- CPPFLAGS :=  -I.. -I../.. $(GDAL_INCLUDE) $(MYSQL_INC) $(CPPFLAGS)
+ CPPFLAGS	:=	-I.. -I../.. $(GDAL_INCLUDE) $(MYSQL_INC) $(CPPFLAGS)
  
 +ifeq ($(MYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION),yes)
 +CPPFLAGS +=   -DMYSQL_NEEDS_LOAD_DEFAULTS_DECLARATION
 +endif
 +
 +
- default: $(O_OBJ:.o=.$(OBJ_EXT))
+ default:	$(O_OBJ:.o=.$(OBJ_EXT))
  
  clean:
 diff --git a/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp b/ogr/ogrsf_frmts/mysql/ogrmysqldatasource.cpp
@@ -430,4 +430,4 @@ index d702594..69ea3c2 100644
 +    #if defined(__APPLE__) && (!defined(TARGET_OS_IPHONE) || TARGET_OS_IPHONE==0)
          #include <crt_externs.h>
          #define environ (*_NSGetEnviron())
-     #else
+     #else
\ No newline at end of file
```

gmisra commented:
Yeah, I've looked at the whitespace and still can't get it to work, including testing your diff above.
At this point, I'm going to cut my losses and move on. If somebody needs the formula for 1.10.x, and they find this PR, I'm happy to help. The formula for 1.11 works just fine.

gmisra commented:
@BrewTestBot test this please

DomT4 commented:
Can you add a `conflicts_with` for the ` gdal ` in Homebrew as they presumably clash? And if you could squash all the ` stable do ` stuff into one block, i.e. near the top with:

```ruby
homepage ""

stable do
  url ""
  sha256 ""

  patch do
    url ""
    sha256 ""
  end
end
```

And similar. After that, I'm happy enough to merge.

gmisra commented:
Alright, made the last set of changes. Should be good to merge now.

DomT4 commented:
Alright, merged in https://github.com/Homebrew/homebrew-versions/commit/c97bd1279706388e503dac4399738da1b9b81383. Please see for future the follow-up style changes I made in https://github.com/Homebrew/homebrew-versions/commit/1285df5c05807f6d46b093a6399c0744b9ab17ef in regards to how things should be ideally formatted. `stable do` blocks should contain the `stable` url and checksum - Since there's also not a `head` stated you could have binned the `stable do` block entirely. `stable do` blocks shouldn't have anything above them that isn't a `homepage`, `head` or `revision`, ideally. Various SHA1s should have been SHA256, etc. Ideally, commits should also be squashed on final submission - Having to squash 15 commits locally was quite something :smile:.

Merged though - I appreciate you sticking through the long review process. Thanks for your contribution to Homebrew, we appreciate it! :smiley_cat:.

gmisra commented:
I know better about squashing commits, my bad. Thanks again for all your help, and everybody who makes homebrew work.
